Code/CryMP/Client/WeatherSystem.cpp

```diff
@@ -0,0 +1,438 @@
+#include "WeatherSystem.h"
+#include "CryCommon/CrySystem/ISystem.h"
+#include "CryCommon/CrySystem/ICryPak.h"
+#include "CryCommon/CrySystem/IConsole.h"
+#include "CryCommon/Cry3DEngine/IMaterial.h"
+#include "CryCommon/CryEntitySystem/IEntitySystem.h"
+#include "CryCommon/CryAction/IVehicleSystem.h"
+#include "CryCommon/CryAction/IGameTokens.h"
+#include "CryCommon/CryMath/Cry_Camera.h"
+#include "CryCommon/CrySystem/ILog.h"
+#include "CrySystem/LocalizationManager.h"
+#include "CrySystem/RandomGenerator.h"
+#include "Library/StringTools.h"
+#include "Library/Util.h"
+#include "Library/WinAPI.h"
+
+#include "ScriptBind_CPPAPI.h"
+#include "Client.h"
+#include "ScriptCommands.h"
+#include "ScriptCallbacks.h"
+#include "DrawTools.h"
+#include "CryGame/GameActions.h"
+#include "CryGame/Actors/Actor.h"
+
+#include <map>
+
+static bool						currently_enabled = false;
+static std::map<int, float[3]>	original_weather_values;
+static std::optional<Vec3>		original_wind;
+
+static std::map<int, string>	active_values;
+static std::set<std::string>	active_effects;
+static std::optional<unsigned>  active_mask;
+
+static constexpr std::array<EERType, 3> static_entities = {
+	eERType_Vegetation,
+	eERType_Brush,
+	eERType_Rope
+};
+
+static const int WEATHER_NAMESPACE		= 2000;
+static const int WEATHER_ENV_NAMESPACE  = 2100;
+
+enum EWeatherVars {
+	WEATHER_ENABLED,
+	WEATHER_WIND,
+	WEATHER_FX,
+	WEATHER_LAYERS,
+	WEATHER_VAR_COUNT
+};
+
+CWeatherSystem::CWeatherSystem() {
+	original_weather_values.clear();
+	m_time = 0;
+	m_lastUpdate = -1000.0f;
+}
+
+void CWeatherSystem::Reset() {
+	RestoreWeather(true);
+	if (original_wind) {
+		gEnv->p3DEngine->SetWind(*original_wind);
+	}
+
+	for (auto& k : active_effects) {
+		if (k[0] == '-') SpawnEffect(k.c_str() + 1);
+		else RemoveEffect(k.c_str());
+	}
+
+	if (active_mask) {
+		RemoveLayer(*active_mask);
+		active_mask.reset();
+	}
+	
+	currently_enabled = false;
+	original_weather_values.clear();
+	active_values.clear();
+	original_wind.reset();
+	active_effects.clear();
+}
+
+void CWeatherSystem::Update(float frameTime) {
+	CServerSynchedStorage *pSSS = g_pGame->GetServerSynchedStorage();
```

Comment:
```suggestion
	CSynchedStorage *pSSS = g_pGame->GetSynchedStorage();
```
Client can't read from server synched storage, probably why changes didnt work including bool/ints, guess you tested on LAN server? Even floats you can use for this, as long as you dont need huge precision,